### ISSUE TITLE
Redundant travels causes lack of pressure in nozzle

### ISSUE THREAD

**Issue by cangelis** [Ultimaker/CuraEngine#679]
Please download the [example project](https://www.dropbox.com/s/ewtv0log8td07sv/unnecessary%20moves.3mf?dl=0)

For example please go to layer `43`, and start moving the nozzle, you will see that even though there is only one wall in that layer, after printing first wall, nozzle goes to start point and travels without extruding as if there is second wall. This causes lack of pressure in the next layer because the material was oozed on the previous layer while traveling without extruding.

**Comment by smartavionics** [comment]
I think it is printing 2 walls but the second wall is very thin (you have compensate wall overlaps enabled).

I recommend that you enable the wall order optimization (experimental section) so that the two wall lines start at the same place which will get rid of a lot of retracts/travels.

![screenshot_2018-01-04_17-52-44](https://user-images.githubusercontent.com/585618/34577148-1b205e86-f178-11e7-86d3-8b221c73844d.png)

**Comment by cangelis** [comment]
Thank you, I also tried disabling compensate wall overlaps, it still generates 2 walls and these 2 walls are overlapped, the resulting surface quality is not good as expected.

Does it suppose to generate only 1 wall? Because the wall thickness of the model is not wide enough to fit two walls.

**Comment by smartavionics** [comment]
> Does it suppose to generate only 1 wall? Because the wall thickness of the model is not wide enough to fit two walls.

No, but one possible workaround is to use spiralize mode and then it will only print 1 wall. You may need to make the model solid for the spiralize to work right.

**Comment by jackha** [comment]
If you want to use the current model, you cannot make the wall a single layer for sure. However you can fiddle around with:
- Horizontal expansion (can also be negative): this will make the model thinner
- Line widths

**Comment by Ghostkeeper** [comment]
Do you happen to have an Outer Wall Inset set to something else than 0?

**Comment by cangelis** [comment]
Thanks for your suggestions, horizontal expansion, increasing/decreasing line width, resizing and  sprialize options work for my case.

If you have this issue in some parts of the model rather than the whole like my example, it is not easy to determine, and if you have a big model, printing it with different parameters is a cumbersome, waste of time and material. 

I really like slic3r's approach for this problem that is just ignoring the overflowing walls. if your line width 0.4mm and the wall is 0.9mm it just prints 2 walls rather than printing 3 overflowing walls or "2 normal"+"1 empty wall".

**Comment by cangelis** [comment]
Because the issue is unpredictable, most of the time, reprinting is required and parameters always need to be tweaked. Some example prints from the same issue.

![a8dc5c9a-86a0-4ce3-b157-ba9c0b3a7910](https://user-images.githubusercontent.com/1162691/35901774-7074c594-0be9-11e8-8b09-c41fb9c53642.jpeg)
![img_20180207_092315](https://user-images.githubusercontent.com/1162691/35901775-709528b6-0be9-11e8-92bc-461d0c0b3f0d.jpg)

**Comment by Ghostkeeper** [comment]
I have that same book, Computational Geometry :smile:

**Comment by Ghostkeeper** [comment]
There was a recent proposal that would say that if the wall overlap compensation reduces the wall to less than 30% or so it would just leave out the whole line.
